cmd/tiller-proxy/main.go

```diff
@@ -180,9 +183,32 @@ func main() {
 		port = "8080"
 	}
 	addr := ":" + port
-	log.WithFields(log.Fields{"addr": addr}).Info("Started Tiller Proxy")
-	err = http.ListenAndServe(addr, n)
-	if err != nil {
-		log.Fatalf("Unable to start the server: %v", err)
+
+	srv := &http.Server{
+		Addr:    addr,
+		Handler: n,
 	}
+
+	go func() {
+		log.WithFields(log.Fields{"addr": addr}).Info("Started Tiller Proxy")
+		err = srv.ListenAndServe()
+		if err != nil {
+			log.Info(err)
+		}
+	}()
+
+	// Catch SIGINT and SIGKILL
```

Comment:
we're actually only handling SIGINT and SIGTERM, Go does not allow us to handle SIGKILL